usng4j-impl/src/main/java/org/codice/usng4j/impl/UsngCoordinateImpl.java

```diff
@@ -183,20 +207,44 @@ private static UsngCoordinate parseCoordinateString(
       throw new ParseException(message, 0);
     }
 
+    int captureGroupsCount = m.groupCount();
     int zoneNumber = Integer.parseInt(m.group(1));
     char latitudeBandLetter = m.group(2).toCharArray()[0];
 
     if (m.group(3) != null) {
       char columnLetter = m.group(3).toCharArray()[0];
       char rowLetter = m.group(3).toCharArray()[1];
 
-      if (!(StringUtils.isEmpty(m.group(4)) || StringUtils.isEmpty(m.group(5)))) {
-        int easting = Integer.parseInt(m.group(4).trim());
-        int northing = Integer.parseInt(m.group(5).trim());
+      if (captureGroupsCount > 4
+          && !(StringUtils.isEmpty(m.group(4)) || StringUtils.isEmpty(m.group(5)))) {
+        String easting = m.group(4).trim();
+        String northing = m.group(5).trim();
+        int eastingInt = Integer.parseInt(easting);
+        int northingInt = Integer.parseInt(northing);
+
+        result =
+            new UsngCoordinateImpl(
+                zoneNumber, latitudeBandLetter, columnLetter, rowLetter, eastingInt, northingInt);
+      } else if (!StringUtils.isEmpty(m.group(4))) {
+        // full numerical location with easting and northing values given as n+n digits
+        String numericalLocation = m.group(4).trim();
+        int splitIndex = numericalLocation.length() / 2;
+        // easting value is first n digits and northing value is last n digits
+        String easting = numericalLocation.substring(0, splitIndex);
+        String northing = numericalLocation.substring(splitIndex);
```

Comment:
Wouldn't Integer.parseInt() drop leading zeros?  Does that cause any issues?

Reply:
That would cause an issue. I made a change to determine the precision of the coordinate before calling `Integer.parseInt()`. This way the precision should be retained throughout the rest of the parsing/conversions and the other math being done on the easting/northing integer values can be left unchanged.